### ISSUE TITLE
S3260 still triggers on some static classes

### ISSUE THREAD

**Issue by mattdg** [SonarSource/sonar-dotnet#5063]
### Description

This was previously reported in issue [4980](https://github.com/SonarSource/sonar-dotnet/issues/4980):
> The rule to seal private classes that are not derived ([S3260](https://sonarsource.github.io/rspec/#/rspec/S3260/csharp)), also triggers on static classes, that cannot be sealed.

However, using `SonarAnalyzer.CSharp` version `8.32.0.39516` still triggers on the case I mentioned in the comments on that previous issue:  non-nested `internal` `static` classes.

### Repro steps

Create a static class that is not nested within another class.

```c#
internal static class SomeStaticClass
{
     // static members
}
```

or this (`internal` is implied)
```c#
static class SomeStaticClass
{
     // static members
}
```

### Expected behavior

No warning from S3260, as `class` is `static`.  Basically, it should ignore ALL `static` `class`es as they don't support inheritance.

### Actual behavior

The analyzer will incorrectly flag this as a violation of [S3260](https://sonarsource.github.io/rspec/#/rspec/S3260/csharp).
> error S3260: Private classes or records which are not derived in the current assembly should be marked as 'sealed'

### Known workarounds

Suppressing false positives in code, or disabling the specific analyzer rule for S3260.

### Related information

* dotnet SDK 5.0.403
* `SonarAnalyzer.CSharp` version `8.32.0.39516`
* Windows 10

**Comment by mary-georgiou-sonarsource** [comment]
Hi @mattdg and thank you for reporting this. 😄 
With the given snippet and env details, I cannot reproduce the issue.
Would it be possible to give me a more detailed reproducer?
Also:
- Do you meet the issue while using some specific framework? 
- How do you build your project? (e.g., Visual  Studio 2019)

Thanks!

**Comment by mattdg** [comment]
I guess this can be closed.  I don't seem to be able to repro this anymore either.  I was able to remove the suppression I had to add after the last update (we compile with warnings as errors).  But it seems to work now without issue.
